myfitnesspal/client.py

```diff
@@ -804,3 +810,545 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+            "nutritional_content[monounsaturated_fat]": "{}".format(
+                monounsaturated_fat
+            ),
+            "nutritional_content[sugar]": "{}".format(sugar),
+            "nutritional_content[trans_fat]": "{}".format(trans_fat),
+            "nutritional_content[protein]": "{}".format(protein),
+            "nutritional_content[cholesterol]": "{}".format(cholesterol),
+            "nutritional_content[vitamin_a]": "{}".format(vitamin_a),
+            "nutritional_content[calcium]": "{}".format(calcium),
+            "nutritional_content[vitamin_c]": "{}".format(vitamin_c),
+            "nutritional_content[iron]": "{}".format(iron),
+            "food_entry[quantity]": "1.0",
+            "food_entry[meal_id]": "0",
+            "addtodiary": "no",
+            "preserve_exact_description_and_brand": "true",
+            "continue": "Save",
+        }
+        # Make entry public if requested, Hint: submit "sharefood": 0 also generates a public db entry, so only add
+        # "sharefood"" if really requested
+        if sharepublic:
+            data["sharefood"] = 1
+
+        url = parse.urljoin(self.BASE_URL_SECURE, SUBMIT_POST_PATH)
+        result = self.session.post(
+            url,
+            data,
+        )
+        if result.status_code == 200:
+            document = lxml.html.document_fromstring(result.content.decode("utf-8"))
+            try:
+                if not document.xpath(
+                    # If list is empty there should be no error, could be replaced with assert
+                    "//*[@id='errorExplanation']/ul/li"
+                ):
+                    return True
+                    # print("No Error :)")
+                    # Would like to return FoodItem, but seems that it take
+                    # to long until the submitted food is available in the DB
+                    # return self.get_food_search_results("{} {}".format(brand, description))[0]
+                else:  # Error occurred
+                    error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
+                    error = error.replace("Description ", "")  # For cosmetic reasons
+                    raise MyfitnesspalRequestFailed(
+                        f"Unable to submit food to MyFitnessPal: {error}"
+                    )
+            except Exception:
+                logger.warning(f"Unable to submit food to MyFitnessPal: {error}")
+
+        elif not result.ok:
+            logger.warning(
+                f"Request Error - Unable to submit food to MyFitnessPal: status code: {result.status_code}"
+            )
+            return None
+
+    def set_new_goal(
+        self,
+        energy: float = "",
+        energy_unit: str = "",
+        carbohydrates: float = "",
+        protein: float = "",
+        fat: float = "",
+        percent_carbohydrates: float = "",
+        percent_protein: float = "",
+        percent_fat: float = "",
```

Comment:
As the the new values will be taken from the old values in case they are not provided it does not make sense here to assign an empty string here. 

I did it because we observed that MFP responded with an error on our post if we sent a None in the payload instead of "".